src/test/java/com/github/dockerjava/cmd/PushImageCmdIT.java

```diff
@@ -50,30 +47,26 @@ public void pushLatest() throws Exception {
         String imageId = dockerRule.getClient().commitCmd(container.getId()).withRepository(username + "/busybox").exec();
 
         // we have to block until image is pushed
-        dockerRule.getClient().pushImageCmd(username + "/busybox").exec(new PushImageResultCallback()).awaitSuccess();
+        dockerRule.getClient().pushImageCmd(username + "/busybox").exec(new PushImageResultCallback()).awaitCompletion();
 
         LOG.info("Removing image: {}", imageId);
         dockerRule.getClient().removeImageCmd(imageId).exec();
 
-        dockerRule.getClient().pullImageCmd(username + "/busybox").exec(new PullImageResultCallback()).awaitSuccess();
+        dockerRule.getClient().pullImageCmd(username + "/busybox").exec(new PullImageResultCallback()).awaitCompletion();
     }
 
     @Test
     public void pushNonExistentImage() throws Exception {
-
-        if (getVersion(dockerRule.getClient())
-                .isGreaterOrEqual(RemoteApiVersion.VERSION_1_26)) {
-         // no errors??
-        } else if (getVersion(dockerRule.getClient())
-                .isGreaterOrEqual(RemoteApiVersion.VERSION_1_24)) {
-            exception.expect(DockerClientException.class);
-        } else {
+        //swarms throws a different error here
+        if (TestUtils.isSwarm(dockerRule.getClient())) {
             exception.expect(NotFoundException.class);
+        } else {
+            exception.expect(DockerClientException.class);
         }
 
         dockerRule.getClient().pushImageCmd(username + "/xxx")
                 .exec(new PushImageResultCallback())
-                .awaitCompletion(20, TimeUnit.SECONDS); // exclude infinite await sleep
+                .awaitCompletion(); // exclude infinite await sleep
```

Comment:
1) comment is wrong
2) without timeout test may stuck forever, test needs some timeout.